### PR TITLE
Add style guide

### PR DESCRIPTION
**Pull request by Mats-SX**
This is a stripped-down version of the style guide that the docs team had previously.

### CODE REVIEW

File: docs/style-guide.adoc [hunk 1]
```diff
@@ -0,0 +1,455 @@
...
+  ON CREATE SET n.prop = 0
+MERGE (a:A)-[:T]-(b:B)
+  ON CREATE SET a.name = 'me'
+  ON MATCH SET b.name = 'you'
+----
+
+.. Put `ON CREATE` before `ON MATCH` if both are present.
+// no example necessary
+
+. Write keywords in all caps.
```
Comment (petraselmer) [inline docs/style-guide.adoc]: I would suggest "write keywords in all caps." -> "Write keywords in upper case."

File: docs/style-guide.adoc [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
++
+.Good
+[source, cypher]
+----
+MATCH (e:Employee)
+RETURN e.name
+----
+
+. Write relationship types in all caps, using underscore to separate words.
++
```
Comment (petraselmer) [inline docs/style-guide.adoc]: I would suggest "write relationship in all caps..." -> "Write relationship types in upper case..."

File: docs/style-guide.adoc [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+.Good
+[source, cypher]
+----
+RETURN 'Cypher'
+----
+
+. *Spacing*
+.. For literal maps:
+... No space between opening brace and first key
```
Comment (petraselmer) [inline docs/style-guide.adoc]: -> "No space between an opening brace and the first key"

File: docs/style-guide.adoc [hunk 4]
```diff
@@ -0,0 +1,455 @@
...
+... No space between key and colon
```
Comment (petraselmer) [inline docs/style-guide.adoc]: -> "No space between a key and colon"

File: docs/style-guide.adoc [hunk 5]
```diff
@@ -0,0 +1,455 @@
...
+... One space between colon and value
+... No space between value and comma
+... One space between comma and next key
+... No space between last value and closing brace
```
Comment (petraselmer) [inline docs/style-guide.adoc]: -> "No space between the last value and the closing brace"

File: docs/style-guide.adoc [hunk 6]
```diff
@@ -0,0 +1,455 @@
...
+----
++
+.Good
+[source, cypher]
+----
+MATCH (:Person)-->(:Vehicle)
+RETURN count(*)
+----
+
+.. Use wrapping space around operators.
```
Comment (petraselmer) [inline docs/style-guide.adoc]: -> "Use a wrapping space around operators."

File: docs/style-guide.adoc [hunk 7]
```diff
@@ -0,0 +1,455 @@
...
+----
++
+.Good
+[source, cypher]
+----
+MATCH (person:Person:Owner)
+RETURN person.name
+----
+
+.. Use space after each comma in lists and enumerations.
```
Comment (petraselmer) [inline docs/style-guide.adoc]: -> "Use a space after each comma in lists and enumerations."

File: docs/style-guide.adoc [hunk 8]
```diff
@@ -0,0 +1,455 @@
...
+.Good
+[source, cypher]
+----
+MATCH (:Person)-->(vehicle:Car)-->(:Company)<--
+      (:Country)
+RETURN count(vehicle)
+----
+
+. Surround node patterns with parentheses.
+.. This only concerns old Cypher code, as it is no longer legal Cypher to omit parentheses.
```
Comment (petraselmer) [inline docs/style-guide.adoc]: -> "This only concerns old Cypher code, as omitting parentheses is no longer legal in Cypher."

File: docs/style-guide.adoc [hunk 9]
```diff
@@ -0,0 +1,455 @@
...
++
+.Good
+[source, cypher]
+----
+MATCH (node:NonSpacedLabel {property: 42})
+RETURN labels(node)
+----
+
+. *Patterns*:
+.. Leave nodes and relationships anonymous when the variable would not be used.
```
Comment (petraselmer) [inline docs/style-guide.adoc]: -> "Use anonymous nodes and relationships when the variable would not be used."

File: docs/style-guide.adoc [hunk 10]
```diff
@@ -0,0 +1,455 @@
...
++
+.Good
+[source, cypher]
+----
+MATCH (manufacturer:Company)<--(vehicle:Car)<--(:Person)
+WHERE manufacturer.founded_year < 2000
+RETURN vehicle.mileage
+----
+
+.. Prefer outgoing (left to right) pattern relationships to incoming.
```
Comment (petraselmer) [inline docs/style-guide.adoc]: -> "Prefer outgoing (left to right) pattern relationships to incoming pattern relationships."

**Comment by Mats-SX** [comment]
@petraselmer Updated, thanks for the review.

**Pull request merged**